### ISSUE TITLE
Not using digital ocean info?

### ISSUE THREAD

**Issue by madasebrof** [billstclair/elm-s3#2]
Hey, so it looks like this doesn't actually check for Digital Ocean? I have an `S3.Type.Account` that's properly set up, but it seems to auto-prepend amazonaws?

I looked through the S3 source code, and the only place `isDigitalOcean` is checked is in the example where the url is manually prepended to the key. The actual S3 library doesn't actually seem to use that anywhere. 

If you I do something like:

```Elm
digitalOceanUrlPrefix =
    "https://nyc3.digitaloceanspaces.com/"

S3.putPublicObject "my_bucket" (digitalOceanUrlPrefix ++ key) body
    |> S3.send Env.spacesAccount
    |> Task.attempt ReceivePutObject
```

It just adds `digitalOceanUrlPrefix` it after the aws address. (If I remove `digitalOceanUrlPrefix`, I just get the straight aws url)

<img width="951" alt="Screen Shot 2022-10-18 at 1 00 24 PM" src="https://user-images.githubusercontent.com/1993403/196497800-454ac60b-f396-4116-9c53-25071b5e82a6.png">

(Note--I get that there is also a CORS problem, but can't fix that until I'm going to the correct URL...)

**Comment by madasebrof** [comment]
Yeah--looks like all built on top of `elm-aws-core`, and am not seeing how to config URL of endpoint in `elm-aws-core`, so not sure this lib can support Digital Ocean?

**Comment by billstclair** [comment]
Weird. I could swear I tested the Digital Ocean code, but apparently not. It looks like `S3.makeService` needs to set the `hostResolver` and (maybe) `regionResolver` properties of the returned `AWS.Service.Service`, when `isDigitalOcean` is true.

The fix probably won't take me very long, but I don't currently have any Digital Ocean spaces set up (they cost $$), so I have to do that to test the change.

I'm finally back to coding a couple of hours a day (after two years away), and I'm adding S3 state support to Mammudeck, so I'm close to that code. If you're feeling competent to do it, let me know by end of day tomorrow, and I'll take your pull request instead of working on it myself. Otherwise, I'll do my best to work on it on Thursday morning, or tomorrow if you let me know by then that you're not doing it (my energy has been unreliable of late; I'm not getting any younger).

**Comment by billstclair** [comment]
I'm up and have some coding chops, so I'm doing this now. I may not finish before I crash tonight, but I'll continue in the morning, God williing and the creek don't rise.

**Comment by billstclair** [comment]
Fixed. Published as version 2.2.0. The "bucket" becomes a URL host subdomain for Digital Ocean, which doesn't really have a concept of bucket. Explained in more detail in the README.

https://package.elm-lang.org/packages/billstclair/elm-s3/latest/

**Comment by madasebrof** [comment]
Wow! Thanks so much!!!
